LoRaEngine/LoraKeysManagerFacade/SearchDeviceByDevEUI.cs

```diff
@@ -52,14 +52,14 @@ private async Task<IActionResult> RunGetDeviceByDevEUI(HttpRequest req)
 
             using var deviceScope = this.logger.BeginDeviceScope(parsedDevEui);
 
-            var device = await this.registryManager.GetDeviceAsync(parsedDevEui.ToString());
-            if (device != null)
+            var primaryKey = await this.registryManager.GetDevicePrimaryKeyAsync(parsedDevEui.ToString());
+            if (primaryKey != null)
```

Comment:
If the device was not found in the Iot hub wouldn't this throw`?

Reply:
As far I know, the SDK doesn't throw an exception when the device doesn't exist. 
To be sure, I just tested what I say, and I can confirm that the GetDevice method returns a null object if not present in the registry.

![image](https://user-images.githubusercontent.com/9513635/182334353-0f0680b5-949f-4dab-8cbe-c2f3bd81f130.png)

Maybe the occasion to create a unit test for that case to clarify this behavior in the wrapper implementation ?

Reply:
As discussed orally, I was not expect the iot hub sdk to throw, rather accessing attributes of a null object. The new changes are fine with me, we could add unit test covering this case to the getdeviceprimarykeyasync that would be neat.